runtime/vm/jfr.cpp

```diff
@@ -1076,10 +1086,10 @@ jfrThreadStatistics(J9VMThread *currentThread)
 	if (NULL != jfrEvent) {
 		initializeEventFields(currentThread, (J9JFREvent *)jfrEvent, J9JFR_EVENT_TYPE_THREAD_STATISTICS);
 
-		jfrEvent->activeThreadCount = vm->totalThreadCount;
-		jfrEvent->daemonThreadCount = vm->daemonThreadCount;
-		jfrEvent->accumulatedThreadCount = vm->accumulatedThreadCount;
-		jfrEvent->peakThreadCount = vm->peakThreadCount;
+		jfrEvent->activeThreadCount = (U_32)vm->totalThreadCount;
+		jfrEvent->daemonThreadCount = (U_32)vm->daemonThreadCount;
+		jfrEvent->accumulatedThreadCount = (U_32)vm->accumulatedThreadCount;
+		jfrEvent->peakThreadCount = (U_32)vm->peakThreadCount;
```

Comment:
This is not appropriate: the types of each pair of fields should be the same (or the left side wider). Suggest the JFR types should be updated to match those of the VM, or upgraded to `U_64`. That goes for both `J9JFRThreadStatistics` and `ThreadStatisticsEntry`.

Also, the return type of `VM_JFRConstantPoolTypes::getRequiredBufferSize()` should be `UDATA`.

Reply:
At first I thought those fields are meant to be 32 bits since we've been following this [website](https://sap.github.io/SapMachine/jfrevents/21.html#javathreadstatistics) for our fields types but the author for `JavaThreadStatistics` event didn't use the correct type so I'll correct types to `U_64`.